internal/strs/strs.go

```diff
@@ -29,99 +29,70 @@ import (
 // It does not care about lowercase or uppercase.
-func IsCamelCase(s string, extraRunes ...rune) bool {
+func IsCamelCase(s string) bool {
 	if s == "" {
 		return false
 	}
 	for _, c := range s {
-		if !((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')) && !containsRune(c, extraRunes) {
+		if !(isLetter(c) || isDigit(c)) {
 			return false
 		}
 	}
 	return true
 }
 
-// IsLowerSnakeCase returns false if s is empty or contains any character that is
-// not between 'a' and 'z' or '0' and '9' or '_', or if s begins or ends
-// with '_'.
-func IsLowerSnakeCase(s string, extraRunes ...rune) bool {
-	if s == "" {
-		return false
-	}
-	if s[0] == '_' {
-		return false
-	}
-	if s[len(s)-1] == '_' {
+// IsLowerSnakeCase returns true if s only contains lowercase letters,
+// digits, and/or underscores. s MUST NOT begin or end with an underscore.
+func IsLowerSnakeCase(s string) bool {
+	if s == "" || s[0] == '_' || s[len(s)-1] == '_' {
 		return false
 	}
-	for _, c := range s {
-		if !((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') || c == '_') && !containsRune(c, extraRunes) {
+	for _, r := range s {
+		if !(isLower(r) || isDigit(r) || r == '_') {
 			return false
 		}
 	}
 	return true
 }
 
-// IsUpperSnakeCase returns false if s is empty or contains any character that is
-// not between 'A' and 'Z' or '0' and '9' or '_', or if s begins or ends
-// with '_'.
-func IsUpperSnakeCase(s string, extraRunes ...rune) bool {
-	if s == "" {
-		return false
-	}
-	if s[0] == '_' {
+// IsUpperSnakeCase returns true if s only contains uppercase letters,
+// digits, and/or underscores. s MUST NOT begin or end with an underscore.
+func IsUpperSnakeCase(s string) bool {
+	if s == "" || s[0] == '_' || s[len(s)-1] == '_' {
 		return false
 	}
-	if s[len(s)-1] == '_' {
-		return false
-	}
-	for _, c := range s {
-		if !((c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_') && !containsRune(c, extraRunes) {
+	for _, r := range s {
+		if !(isUpper(r) || isDigit(r) || r == '_') {
 			return false
 		}
 	}
 	return true
 }
 
-// IsLowercase returns true if s is not empty and is all lowercase.
-func IsLowercase(s string) bool {
-	if s == "" {
-		return false
-	}
-	return strings.ToLower(s) == s
-}
-
-// IsUppercase returns true if s is not empty and is all uppercase.
-func IsUppercase(s string) bool {
-	if s == "" {
-		return false
-	}
-	return strings.ToUpper(s) == s
-}
-
 // ToUpperSnakeCase converts s to UPPER_SNAKE_CASE.
 func ToUpperSnakeCase(s string) string {
-	return strings.ToUpper(ToSnakeCase(s))
+	return strings.ToUpper(toSnake(s))
 }
 
-// ToSnakeCase converts s to Snake_case.
+// toSnake converts s to snake_case.
 // It is assumed s has no spaces.
-func ToSnakeCase(s string) string {
+func toSnake(s string) string {
 	output := ""
 	for i, c := range s {
 		if i > 0 && unicode.IsUpper(c) && output[len(output)-1] != '_' && i < len(s)-1 && !unicode.IsUpper(rune(s[i+1])) {
```

Comment:
We did `unicode.IsUpper` in #50, we should probably change it back.

Reply:
Oops, nice catch! Meant to do that here as well.